### ISSUE TITLE
Native compiler is unable to handle disallowed file paths in import statements

### ISSUE THREAD

**Issue by snario** [EthWorks/Waffle#60]
I think there is a flag you can pass to `solc` called `--allow-paths` that fixes an error I'm facing:

```
contracts/TicTacToeApp.sol:4:1: ParserError: Source "/Users/liam/src/code/monorepo/node_modules/@counterfactual/contracts/contracts/libs/Transfer.sol" not found: File outside of allowed directories.
import "@counterfactual/contracts/contracts/libs/Transfer.sol";
^-------------------------------------------------------------^
```

The string gets correctly mapped to place where the file is located, but `solc` rejects it.

Google results from the error: https://www.google.com/search?q=File+outside+of+allowed+directories.&rlz=1C5CHFA_enCA715CA715&oq=File+outside+of+allowed+directories.&aqs=chrome..69i57j69i64.438j0j1&sourceid=chrome&ie=UTF-8

**Comment by marekkirejczyk** [comment]
While the source of the problem might be with allowed-paths, my first suspect would be a properly setup npm path. Might be tricky with monorepo. 

You can use as example UniversalLogin setup here: https://github.com/UniversalLogin/UniversalLoginSDK/blob/master/universal-login-contracts/.waffle.js

**Comment by marekkirejczyk** [comment]
Ok, I was able to reproduce. Not sure yet what the problem is. Will work on it.

**Comment by marekkirejczyk** [comment]
I implemented the additional allowedPaths feature.
TBH didn't test it live (just write some tests). Let me know if it is helpful.

Documentation [here](https://ethereum-waffle.readthedocs.io/en/latest/configuration.html#lernajs-native-solc).
PR [here](https://github.com/EthWorks/Waffle/pull/63). 
Released in [2.0.1](https://github.com/EthWorks/Waffle/releases/tag/2.0.1)

Closing for now. Feel free to reopen if buggy or not helpful :)
